### ISSUE TITLE
Add support for PostgreSQL in QAN

### ISSUE THREAD

**Issue by gordan-bobic** [shatteredsilicon/ssm-submodules#121]
Add support for PostgreSQL in Query Analytics, including:

1) Harvesting queries from the log
2) Showing performance metrics
3) Showing `EXPLAIN` plans
4) Showing table definitions

The presentation format should be kept as similar as possible to MySQL/MariaDB QAN, but due to differences in the way PostgreSQL does things, for example:

1) There is, AFAIK, no way to add indexes directly within the `CREATE TABLE` statement, they have to be added using `CREATE INDEX` statements afterwards. That means we cannot present a simple CREATE TABLE statement like we do with MySQL, we will need `CREATE TABLE` + `CREATE INDEX` statements for each table.

2) EXPLAIN output is quite different from all of the formats MySQL provides, so that section will have to be different for PostgreSQL.

That means that the QAN page will have to switch layout dynamically based on whether the server is MySQL/MariaDB or PostgreSQL.

**Comment by gordan-bobic** [comment]
Regarding PostgreSQL QAN, here's a few useful bits of info for enabling query logging, similar to MySQL's slow log:
```
log_min_duration_statement = 1000 # Log queries taking longer than 1000ms (1 second)
log_duration = on                 # Logs duration of all statements (optional)
log_statement = 'all'             # Logs all statements; set to 'ddl' or 'mod' for specific types (optional)
```
Also:
```
logging_collector = on
log_destination = 'csvlog'  # Or 'stderr'
log_directory = 'pg_log'    # Directory for log files
log_filename = 'postgresql-%Y-%m-%d_%H%M%S.log'  # Log file naming pattern
```
There is also an equivalent for logging the explain output - use auto_explain for Query Plans:
```
shared_preload_libraries = 'auto_explain'
auto_explain.log_min_duration = 1000  # Log plans for queries taking > 1000ms
auto_explain.log_analyze = on         # Include actual execution stats
```
There is also an equivalent to harvesting performance_schema in MySQL, based on the pg_stat_statements extension:
```
shared_preload_libraries = 'pg_stat_statements'
pg_stat_statements.track = all  # Tracks all statements (or 'top' for top-level, 'none' to disable)
pg_stat_statements.max = 10000  # Max number of queries to store (adjust as needed)
```
The queries can then be harvested from the pg_stat_statements view:
Example query:
```
SELECT
    query,
    calls,
    total_exec_time,
    mean_exec_time,
    rows,
    shared_blks_hit,
    shared_blks_read
FROM pg_stat_statements
WHERE total_exec_time > 1000  -- Filter queries taking > 1000ms
ORDER BY total_exec_time DESC
LIMIT 10;
```
Key columns:
- query: Normalized query text (parameters replaced with placeholders).
- calls: Number of times the query was executed.
- total_exec_time: Total execution time (in milliseconds).
- mean_exec_time: Average execution time per call.
- rows: Total rows affected or retrieved.
- shared_blks_hit/shared_blks_read: Cache hits and disk reads.

**Comment by gordan-bobic** [comment]
Note: We need this feature to support PostgreSQL 13+, v13 is the oldest version that is not currently EOL. Everything older we don't need to be concerned with.

**Comment by oblitorum** [comment]
OK, the QAN page for PostgreSQL instances looks like this (there are some obvious minor bugs I will need to fix):

<img width="3360" height="7132" alt="Image" src="https://github.com/user-attachments/assets/b42e9be6-9a21-4758-9e2c-9d2a848d5e22" />

What do you think?

And it turns out when you set `log_duration = on` and `log_statement = 'all'` at the same time, it separates the statement and duration into 2 log entries:

```
...
2025-07-16 21:04:35.622 UTC [170598] root@dvdrental LOG:  statement: SELECT * FROM public.address WHERE address_id = 5;
2025-07-16 21:04:35.642 UTC [170598] root@dvdrental LOG:  duration: 23.323 ms
...
``` 

Which I think will be a problem when there are multiple connections running queries at the same time, it will mess up the order and make it hard to tell which duration is associated to which statement. So I think for log file harvesting to work, you must set `log_duration = on` and `log_statement = 'none'`, this makes duration and statement go to same log entry:

```
...
2025-07-16 21:14:05.130 UTC [171229] root@dvdrental LOG:  duration: 3.413 ms  statement: SELECT * FROM public.address WHERE address_id = 5;
...
```

Is this  (must set `log_duration = on` and `log_statement = 'none'` for log file harvesting to work) acceptable?

And it seems table harvesting actually has more metric data than log file harvesting, so maybe we should make table harvesting as the default method for PostgreSQL?

**Comment by gordan-bobic** [comment]
That looks really good - very similar to the MySQL/MariaDB QAN layout, which is exactly what we want.
Interesting, so `log_statement = 'none'` still results in the statement being logged? I didn't expect that but I never tried that combination. But yes, that is totally acceptable if it gives us what we need.

Because PostgreSQL doesn't have syntax that allows indexes to be part of the CREATE TABLE statement, can you make sure the INDEXES section is present like in the MySQL / MariaDB case?

And yes, I'm OK with logsource=pg_stat_statements being the default - **if** pg_state_statements plugin is loaded.

**Comment by oblitorum** [comment]
> Interesting, so log_statement = 'none' still results in the statement being logged?

Yeah, `log_statement = 'none'` means no statment log entry. But if you have `log_duration = on` set, and when it tries to log the duration log entry, it will check whether statement was logged, and will log it if it wasn't, which is indeed weird... 

> can you make sure the INDEXES section is present like in the MySQL / MariaDB case?

Yeah, I will add these `CREATE INDEX ...` statements right below the `CREATE TABLE ...` statement.

**Comment by gordan-bobic** [comment]
OK, but we still need the INDEXES section for index cardinality stats.

**Comment by oblitorum** [comment]
Oh, you meant the index stats, will do that too.

**Comment by oblitorum** [comment]
OK, added the index section and the status section:

<img width="3360" height="7240" alt="Image" src="https://github.com/user-attachments/assets/018cd387-1c0d-48af-9010-eb9a9bc7761f" />
